docs/src/developing/backwards-compatibility.md

```diff
@@ -155,3 +165,81 @@ circumvented in order to rapidly deploy a fix, depending on the severity of the
 CLI tooling json output (`output --json`) compatibility will be preserved; however, output directed
 for a human reader is subject to change. This includes output as well as potential help, warning, or
 error messages.
+
+### Why not just use SemVer
+
+The Solana Rust SDK crates do *not* follow semantic versioning (SemVer) for
+breaking changes.
+
+#### SemVer in Rust
+
+Under SemVer, breaking changes, such as removal of functions or types, only happens
+in a new major version.
+
+In many situations, this is useful -- it is undesirable to pick up a breaking
+change accidentally. The default dependency declaration in Cargo follows SemVer.
+It assumes that all minor versions are compatible, and will automatically update
+to the newest minor version available.
+
+For example, if you declare `solana-program = "1.10"`, Cargo can pull in version
+`1.16`, since it assumes that all `1.X.Y` releases are compatible.
+
+If, however, Cargo deems that two versions of a package are incompatible, it will
+treat them as two completely separate packages.
+
+Packages can be deemed incompatible for many reasons, but the most common situation
+under the default declaration format (SemVer) is due to different major versions.
+
+If a package ends up with two or more instances of `solana-program`,
+[bad things will happen](https://doc.rust-lang.org/cargo/reference/resolver.html#version-incompatibility-hazards),
+either at compile-time or runtime, if the conflicting types in the public APIs
+are ever used together.
```

Comment:
Could you explain why [the Semver trick](https://github.com/dtolnay/semver-trick) cannot be used to re-export types in the event of a 2.x release?  It seems as though that would work.  My initial assumption is that there would be too many types to deal with.

This page links to the "version incompatibility hazards" section of the cargo reference, but the cargo reference indicates a solution.  So imho this section should say why that solution will not work in this case.